### PR TITLE
Unify the end request, form data and error events.

### PR DESCRIPTION
**Pull request by nblumhardt**
Rather than a number of disparate events with varied information, adds `FormData` (if enabled) and an error level/`Exception` to the timing-enabled request logging event.

### CODE REVIEW

File: src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs [hunk 1]
```diff
@@ -159,41 +158,35 @@ public void Init(HttpApplication application)
...
+                    var logger = Logger;
+                    if (ShouldLogFormData)
                     {
                         var form = request.Unvalidated.Form;
-
                         if (form.HasKeys())
                         {
                             var formData = form.AllKeys.SelectMany(k => (form.GetValues(k) ?? new string[0]).Select(v => new { Name = k, Value = FilterPasswords(k, v) }));
-                            Logger.Write(_formDataLoggingLevel, "Client provided {@FormData}", formData);
+                            logger = logger.ForContext("FormData", formData, true);
```
Comment (mattgwagner) [inline src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs:171]: Ah, that's a good thought!

File: src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs [hunk 2]
```diff
@@ -159,41 +158,35 @@ public void Init(HttpApplication application)
                     if (request == null)
                         return;
 
-                    Logger.Write(_requestLoggingLevel, "HTTP {Method} {RawUrl} responded {StatusCode} in {ElapsedMilliseconds}ms", 
-                        request.HttpMethod, 
-                        request.RawUrl, 
-                        application.Response.StatusCode, 
-                        stopwatch.ElapsedMilliseconds);
+                    var error = application.Server.GetLastError();
```
Comment (mattgwagner) [inline src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs:161]: I see what you meant on the other PR now, this does make it a bit clearer.

File: src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs [hunk 3]
```diff
@@ -159,41 +158,35 @@ public void Init(HttpApplication application)
...
             };
         }
 
-        static bool ShouldLogFormData()
-        {
-            return Logger.IsEnabled(_formDataLoggingLevel) 
-                && (LogPostedFormData == LogPostedFormDataOption.Always
-                || (LogPostedFormData == LogPostedFormDataOption.OnlyOnError && HttpContext.Current.Response.StatusCode >= 500));
-        }
+        static bool ShouldLogFormData => Logger.IsEnabled(_formDataLoggingLevel) &&
```
Comment (mattgwagner) [inline src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs:187]: I haven't seen this syntax before, is this with C# 6?

Comment (nblumhardt) [inline src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs:187]: Yes, I'm just digging into it myself and finding some nice little conveniences such as this one.

**Pull request merged**